src/dict.h

```diff
@@ -57,6 +57,7 @@ typedef struct dictEntry {
 
 typedef struct dictType {
     uint64_t (*hashFunction)(const void *key);
+    int (*expandCheck)(size_t moremem, double ratio);
```

Comment:
The name seems ok to me.
maybe `expandAllowed` is better (considering the boolean return value)?
maybe rename `ratio` to `usedRatio`?

p.s. i would rather put this member last. this way if someone misses the warning on old dictType instantiations the member will be set to NULL (i.e. the warning can in theory be ignored).

Reply:
Thanks @oranagra I have modified somethings according to your suggestions